src/frontends/lean/builtin_cmds.cpp

```diff
@@ -403,13 +411,22 @@ static environment eval_cmd(parser & p) {
     auto out = p.mk_message(p.cmd_pos(), p.pos(), INFORMATION);
     out.set_caption("eval result");
     scope_traces_as_messages scope_traces(p.get_stream_name(), p.cmd_pos());
-    char * redirected; size_t redir_size;
-    FILE * fp = open_memstream(&redirected, &redir_size);
-    std::unique_ptr<char *, stream_buffer_delete> stream_buffer(&redirected);
+    object_ref pipe(lean_io_mk_pipe(box(0)));
```

Comment:
We must use `io_mk_world()` instead of `box(0)`. Otherwise, we are screwed if we change how we represent `IO`.